docs/source/contributing.rst

```diff
@@ -4,88 +4,139 @@ Contributing
-detailed below.
+members of the community. If you haven't read our :doc:`code of conduct<conduct>`,
+do so here. Manim is free and open source software (FOSS) for mathematical
+animations. As such, **we welcome everyone** who is interested in
+mathematics, pedagogy, computer animations, open-source,
+software development, and beyond. Manim accepts many kinds of contributions,
+some are detailed below:
 
-Many ways of contributing will involve writing, reading, testing, or
-refactoring code.  As our repository is a Fork of `Manim by
-3b1b <https://github.com/3b1b/manim>`_, contributing in this way can be a bit
-confusing.  Here is a short guide on how to do it.
+*  Code maintenance and development
+*  DevOps
+*  Documentation
+*  Developing educational content & narrative documentation
+*  Plugins to extend Manim functionality
+*  Testing (graphical, unit & video)
+*  Website design and development
 
+Contributing can be confusing, so here's a guide:
 
-Setup Manim and version control
+
+Manim Development Process
 ===============================
 
-#. *Make a fork of this repository on Github.*
-   You will need an account with Github. This will allow you to make pull requests (PR)
+For first-time contributors
+----------------------------
+#. Install git:
+
+   For instructions see https://git-scm.com/.
+
+
+#. Fork the project. Go to https://github.com/ManimCommunity/manim and
+   click the "fork" button to create your own copy of the project. You will
+   need a GitHub account. This will allow you to make a "pull request" (PR)
    to the ManimCommunity repo later on.
 
-#. *Clone your fork.*
-   You can clone your Github fork by running:
+#. Clone your fork to your local computer:
 
    .. code-block:: shell
 
-      git clone <your-fork-url>
-      cd manim
+      git clone https://github.com/<your-username>/manim.git
 
    GitHub will provide both a SSH (``git@github.com:<your-username>/manim.git``) and
-   HTTPS (``https://github.com/<your-username>/manim.git``) URL for cloning the repo.
-   You can use whichever one you are setup for.
+   HTTPS (``https://github.com/<your-username>/manim.git``) URL for cloning.
+   You can use SSH if you have SSH keys setup.
 
    .. WARNING::
 
-      Do not ``git clone`` the original ManimCommunity repository.  You must
-      clone your own fork.  After this step, there are three different
-      repositories to keep track of: the original ManimCommunity repo, your own
-      fork of it, and your local repository.
+      Do not clone the ManimCommunity repository. You must clone your own
+      fork.
+
+#.  Change the directory to enter the project folder:
+
+    .. code-block:: shell
+
+       cd manim
 
-#. Make ``git`` aware of the ManimCommunity repo.
+#. Add the upstream repository, ManimCommunity:
 
    .. code-block:: shell
 
       git remote add upstream https://github.com/ManimCommunity/manim.git
-      git fetch upstream
```

Comment:
user's should also fetch the repository or they can't create branches from it.

Reply:
I mainly followed NumPy's documentations, but down [below ](https://github.com/ManimCommunity/manim/pull/1130/files/e9fe12bb885f1cfa0d32ee67c1cf63a48b25613c#r598280231), once they've `git pull upstream master` they can create branches -- since `git pull` is essentially `git fetch + merge`

Reply:
`git pull` usually creates a merge commit and that isn't great

Reply:
https://github.com/ManimCommunity/manim/pull/1130/files/e9fe12bb885f1cfa0d32ee67c1cf63a48b25613c#r598282780

Reply:
The `git pull upstream master` should be done off the local master, so assuming good practices, the user will **only** ever do fast-forward merges

Reply:
fine over here then
but still I would like people to use branches rather than committing on master branch, in a way people don't need to update the master branch itself.

Reply:
Agreed. This is why I chose not to tell users to do `git checkout -b master upstream/master` and instead do `git checkout -b <new branch name>` from their master (see line 104)